Lib/test/test_array.py

```diff
@@ -1665,5 +1665,12 @@ def test_tolist(self, size):
         self.assertEqual(ls[:8], list(example[:8]))
         self.assertEqual(ls[-8:], list(example[-8:]))
 
+    def test_gh_128961(self):
+        a = array.array('i')
+        it = iter(a)
+        list(it)
+        it.__setstate__(0)
+        self.assertRaises(StopIteration, lambda: next(it))
```

Comment:
You can drop the lambda ;) Moreover, let's exploit this opportunity to add an extra newline before `if __name__ == "__main__":`.

```suggestion
        self.assertRaises(StopIteration, next, it)

```